internal/pkg/cli/job_delete.go

```diff
@@ -164,6 +234,110 @@ func (o *deleteJobOpts) askJobName() error {
 	return nil
 }
 
+func (o *deleteJobOpts) appEnvironments() error {
+	if o.envName != "" {
+		env, err := o.targetEnv()
+		if err != nil {
+			return err
+		}
+		o.environments = append(o.environments, env)
+	} else {
+		envs, err := o.store.ListEnvironments(o.appName)
+		if err != nil {
+			return fmt.Errorf("list environments: %w", err)
+		}
+		o.environments = envs
```

Comment:
nit: We probably do this in the older commands, but I think it's slightly more readable if we don't store the environments in the state and just return them: 
```suggestion
		return envs, nil
```
this way it becomes clear in subsequent methods the data that's needed to for executing them.
For example: `deleteJobStacksFrom(envs)`. 

A rule-of-thumb can be:
1. Store flag values in the struct (*Vars)
2. Store client dependencies in the struct

The rest of the data should be just passed around

Reply:
Gotcha. Thanks! Changed in `job_delete` and `svc_delete`.